client/evm_contract.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/loomnetwork/go-loom"
+	"github.com/loomnetwork/go-loom/auth"
+	"github.com/loomnetwork/go-loom/vm"
+)
+
+// Contract provides a thin abstraction over DAppChainClient that makes it easier to perform
+// read & write operations on a contract running on a Loom DAppChain.
+type EvmContract struct {
+	client  *DAppChainRPCClient
+	Address loom.Address
+	Name    string
+}
+
+func NewEvmContract(client *DAppChainRPCClient, contractAddr loom.LocalAddress) *EvmContract {
+	return &EvmContract{
+		client: client,
+		Address: loom.Address{
+			ChainID: client.GetChainID(),
+			Local:   contractAddr,
+		},
+	}
+}
+
+func NewDeployEvmContract(client *DAppChainRPCClient, signer auth.Signer, byteCode []byte, name string) *EvmContract {
```

Comment:
`NewDeployEvmContract` -> `DeployContract`... and don't swallow the errors, return them.